### PR TITLE
"Factories, new resources, NPC Strongholds" update support

### PR DESCRIPTION
**Pull request by truekabal**
- Added new update constants and new entities support
- Removed EnergyContainer interface. Improved Store interface instead.
- Minor structural improvements

New Store interface has some interesting features.
For now all entities, which had energy, power or any other resource in their entity variable, implement Store instead.

If getCapacity(), getFreeCapacity(), getUsedCapacity() is called by entity which can contain any resource (creep, Storage, Terminal, Container) - overall capacity will be returned.
**But**!!  getCapacity() and other methods which would be called by for example Link or Spawn will return null, because you need to set the exact ResourceConstant _(getCapacity(RESOURCE_ENERGY))_ for getting max capacity for that resource

Assuming that I'd be glad if someone improve Store interface for all possible variants of usage

### CODE REVIEW

File: src/main/kotlin/screeps/api/Constants.kt [hunk 1]
```diff
@@ -4,6 +4,7 @@ package screeps.api
 
 import screeps.api.structures.Structure
 import screeps.api.structures.StructureSpawn
+import kotlin.contracts.Effect
```
Comment (MaggNorway) [inline src/main/kotlin/screeps/api/Constants.kt]: I guess this isn't needed? :)

Comment (truekabal) [reply src/main/kotlin/screeps/api/Constants.kt]: Yeah, you're right. I'll remove it this evening

**Comment by MaggNorway** [comment]
I've added you to the #kotlin channel on the Screeps Slack

**Comment by truekabal** [comment]
Pushed code review fixes.
Also fixed Ruin and Tombstore errors.

I decided to rename IStore to StoreOwner and StoreDefinition to Store. As for me this names show exact definition of interface

**Comment by exaV** [comment]
Thanks @truekabal. I like the new names a lot better :)
I will check why the build is not working

File: src/main/kotlin/screeps/api/Ruin.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package screeps.api
+
+abstract external class Ruin : RoomObjectNotNull, Decaying, Identifiable, IStore {
+    val creep: Creep
```
[the remark this answers is not in the document]
Comment (truekabal) [reply src/main/kotlin/screeps/api/Ruin.kt]: yes, it should and it is already.
You're watching not the last version of this file in pull request

**Comment by exaV** [comment]
Thanks!

**Pull request merged**